### PR TITLE
Use StopAPI from mobility-toolbox-js in StopFinder

### PR DESCRIPTION
**Pull request by edellucien**
# How to

Use StopAPI from mobility-toolbox.js

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [ ] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [ ] IE11 tested.
- [ ] Labels applied. if it's a release? a hotfix?
- [ ] Tests added.

### CODE REVIEW

File: src/components/Search/engines/StopFinder.js [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
-    return fetch(
-      `${this.endpoint}?&q=${encodeURIComponent(value)}&key=${
-        this.apiKey
-      }&${optionsString}`,
-    )
-      .then((data) => data.json())
-      .then((featureCollection) => featureCollection.features);
+    return this.api.search({
+      q: value,
+    });
```
Comment (oterral) [inline src/components/Search/engines/StopFinder.js:19]: What happens with the use of options and the encodeURIComponnet function ? If they were there it was for a resaon I guess

**Review by oterral — CHANGES_REQUESTED** [review]
Did you test with trafimage  and other ?

File: src/components/Search/engines/StopFinder.js (hunk 1 above)
Comment (edellucien) [reply src/components/Search/engines/StopFinder.js:19]: Indeed forgot that! Thanks

Comment (edellucien) [reply src/components/Search/engines/StopFinder.js:19]: If that is it, we could merge it.

File: src/components/Search/Search.js [hunk 2]
```diff
@@ -70,11 +70,21 @@ function Search({
...
   className,
 }) {
+  const currentEngines = useMemo(() => {
+    if (!engines) {
+      return {
+        stops: new StopFinder(null, { apiKey }),
+      };
+    }
+    return engines;
+  }, [apiKey, engines]);
```
Comment (oterral) [inline src/components/Search/Search.js:80]: Does this code could be in the useMemo of SearchService ?

File: src/components/Search/engines/StopFinder.js (hunk 1 above)
Comment (oterral) [reply src/components/Search/engines/StopFinder.js:19]: and options ?

Comment (edellucien) [reply src/components/Search/engines/StopFinder.js:20]: Of course you are right, options are added as well

**Review by oterral — APPROVED** [review]

**Pull request merged**